packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py

```diff
@@ -160,25 +187,14 @@ async def _set(self, new_value: ItemConcreteValue) -> None:
             # convert the concrete value to a data value
             converted_value: ItemConcreteValue = self._py_value_converter(new_value)
 
-            def _raise_if_file_not_on_disk(path: Path) -> None:
-                if not path.exists() or not path.is_file():
-                    log.warning("Could not find a valid file %s", path)
-                    raise InvalidItemTypeError(self.property_type, str(new_value))
-
             if isinstance(converted_value, Path):
-                if converted_value.is_symlink():
-                    symlink_target_path = Path(os.readlink(converted_value))
-                    if not symlink_target_path.is_absolute():
-                        # in the case of a relative symlink
-                        # since the working directory is different form the
-                        # one in which the symlink is placed,
-                        # it must be transfromed to an absolute path
-                        symlink_target_path = (
-                            converted_value.parent / symlink_target_path
-                        )
-                    _raise_if_file_not_on_disk(symlink_target_path)
-                else:
-                    _raise_if_file_not_on_disk(converted_value)
+                if (
+                    not port_utils.is_file_type(self.property_type)
+                    or not converted_value.exists()
+                    or converted_value.is_dir()
+                ):
+                    raise InvalidItemTypeError(self.property_type, f"{new_value}")
+                _raise_if_symlink_not_valid(converted_value)
```

Comment:
I think it would be more explicit if you do something like:
```python
if converted_value.is_symlink():
  _raise_if_symlink_not_valid(converted_value)
```